src/common/src/consistent_hashing.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::collections::HashMap;
+
+use risingwave_pb::common::VNodeRanges;
+
+use crate::hash::{VirtualNode, VIRTUAL_NODE_COUNT};
+
+pub type ParallelUnitId = u32;
+
+pub fn full_vnode_range() -> VNodeRanges {
+    VNodeRanges {
+        starts: vec![0],
+        ends: vec![VIRTUAL_NODE_COUNT as u64 - 1],
+    }
+}
+
+#[expect(clippy::type_complexity)]
+pub fn build_vnode_mapping(
+    parallel_units: &[u32],
+) -> (
```

Comment:
We may not use vnode mappings built here. Instead, we have to use vnode mappings built in meta when an mv is created. I'll implement this (inform frontend of the vnode mappings) soon.